src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs

```diff
@@ -452,131 +568,61 @@ impl WingAntiIceComplex {
+        self.wai_system_has_fault = false;
 
         self.wai_relay.update(context, wai_mode);
 
-        for n in 0..NUM_OF_WAI {
-            self.update_pressure_above_minimum(
-                context,
-                engine_systems[n].precooler_outlet_pressure(),
-                n,
-            );
-
-            // First, we see if the valve's open amount changes this update,
-            // as a result of a change in the ovhd panel push button.
-            // If the precooler is not pressurized, a FAULT should light.
-            self.update_valve_controller(context, n);
-            self.wai_valve[n].update_open_amount(&self.wai_valve_controller[n]);
-
-            // We need both controllers to signal `on` for the
-            // system to be considered on without a fault.
-            if self.wai_valve_controller[n].controller_signals_on() {
-                // If a controller signals `on` while its corresponding valve is closed
-                // this means the system has a fault.
-                if self.is_wai_valve_closed(n) {
-                    has_fault = true;
-                }
-            }
-
-            // An exhaust tick always happens, no matter what
-            // the valve's state is
-            self.wai_exhaust[n].update_move_fluid(context, &mut self.wai_consumer[n]);
-
-            // The heated slats radiate energy to the ambient atmosphere.
-            self.wai_consumer[n].radiate_heat_to_ambient(context);
-
-            // This only changes the volume if open_amount is not zero.
-            self.wai_valve[n].update_move_fluid_with_transfer_speed(
-                context,
-                &mut engine_systems[n].precooler_outlet_pipe,
-                &mut self.wai_consumer[n],
-                Self::WAI_VALVE_TRANSFER_SPEED,
-            );
-
-            if self.wai_consumer_pressure(n)
-                <= Pressure::new::<bar>(Self::WAI_MIN_PRESSURE) + context.ambient_pressure()
-                && !self.is_wai_valve_closed(n)
-            {
-                self.wai_high_pressure[n] = false;
-                self.wai_low_pressure[n] = true;
-                has_fault = true;
-            } else if self.wai_consumer_pressure(n)
-                >= Pressure::new::<bar>(Self::WAI_MAX_PRESSURE) + context.ambient_pressure()
-                && !self.is_wai_valve_closed(n)
-            {
-                self.wai_high_pressure[n] = true;
-                self.wai_low_pressure[n] = false;
-                // High pressure doesn't turn the FAULT light ON
-                // has_fault = true;
-            } else {
-                self.wai_high_pressure[n] = false;
-                self.wai_low_pressure[n] = false;
-            }
+        for (wai_system, engine_system) in self.wai_systems.iter_mut().zip(engine_systems) {
+            wai_system.update(context, engine_system, &self.wai_relay);
+            self.wai_system_has_fault = self.wai_system_has_fault || wai_system.wai_has_fault();
```

Comment:
Just a small note
```suggestion
            self.wai_system_has_fault ||= wai_system.wai_has_fault();
```

Reply:
I thought about that! But at least on my computer, Rust doesn't support ||=

Reply:
You're right there is no ||= operator in Rust. My bad.